runtime/lua/vim/version.lua

```diff
@@ -423,8 +421,12 @@ function M.parse(version, opts)
 end
 
 setmetatable(M, {
+  --- Returns the current Nvim version.
   __call = function()
-    return vim.fn.api_info().version
+    local version = vim.fn.api_info().version
+    -- Workaround: vim.fn.api_info().version reports "prerelease" as a boolean.
+    version.prerelease = version.prerelease and 'dev' or nil
```

Comment:
Hack: just return "dev" for now, so that at least the prerelease field is properly a string. Later maybe someone can find a way to get the actual prerelease string.